platform/qt/src/string_stdlib.cpp

```diff
@@ -9,13 +9,13 @@ namespace mbgl {
 namespace platform {
 
 std::string uppercase(const std::string& str) {
```

Comment:
> The context is we should never use c_str() for multibyte.

That is interesting -- I thought the only difference was that `data()` could leave off null termination (and probably for most of our compiler targets it null terminates anyway, which makes it easy to introduce sneaky bugs depending on the null termination).